src/posthog-core.ts

```diff
@@ -348,6 +351,13 @@ export class PostHog {
 
         this._gdpr_init()
 
+        if (config.segment) {
+            config.bootstrap = config.bootstrap || {}
+            config.bootstrap.distinctID = config.segment.user().id() || config.segment.user().anonymousId() || undefined
```

Comment:
this bit is tricky, as if the distinctID is not `config.bootstrap.distinctID`, the feature flags coming from bootstrap (line 370 below) are invalidated, as the ID has changed.

Will need to go in the docs as well, whatever we decide here. I guess we can not bootstrap flags if the IDs aren't the same 🤷 . Or disable bootstrapping flags completely when using the segment integration.

Reply:
Not super sure what this means or what a good solution is. Any hints? 😅

Reply:
Decided to just avoid using bootstrapping altogether. Realised we have a way of overriding the anonymousId so that solves all the issues anyways